govcd/system.go

```diff
@@ -999,5 +1002,8 @@ func QueryOrgVdcStorageProfileByID(vcdCli *VCDClient, id string) ([]*types.Query
 	if err != nil {
 		return nil, err
 	}
-	return results.Results.OrgVdcStorageProfileRecord, nil
+	if len(results.Results.OrgVdcStorageProfileRecord) != 1 {
```

Comment:
I hope you don't mind, but one more spot.

There is a particular sentinel error definition `ErrorEntityNotFound` in this SDK. When there are no results (and we look for an exact thing - we do return `ErrorEntityNotFound`. On the other hand if there are more than one results (which is unlikely in this case because you filter by ID - we do not return such error.

This helps in code to handle a specific error case (no element found instead of some connection, malformed id, etc. error). To ease consumption of such error two additional helper functions are in place for validation - `ContainsNotFound` and `IsNotFound`. 


In some cases it is a very convenient way to then write check like that:

```
if ContainsNotFound(err) { // internally it validates if err != nil
   // handle unavailable item
}

if err != nil { // Some other error occured and must be handled differently.

}
```

Excerpt from `QueryVmById`
```
	if len(foundVM) == 0 {
		return nil, ErrorEntityNotFound
	}
	if len(foundVM) > 1 {
		return nil, fmt.Errorf("more than one VM found with ID %s", id)
	}
```

Reply:
no problems, fixed.